sklearn/utils/tests/test_param_validation.py

```diff
@@ -157,12 +158,12 @@ def test_generate_invalid_param_val(constraint):
 @pytest.mark.parametrize(
     "constraint",
     [
-        _ArrayLikes,
-        _Callables,
-        _InstancesOf,
-        _NoneConstraint,
-        _RandomStates,
-        _SparseMatrices,
+        _ArrayLikes(),
```

Comment:
How come this was working previously?

I think that the documentation of `generate_invalid_param_val` should be updated to mention that we expect a `Constraint instance` and not only `Constraint`.

Reply:
It used to work because it did raise a NotImplementedError which is what the test is checking. I added that an instance is expected